teeworlds7/user_actions.go

```diff
@@ -76,34 +85,49 @@ func (client *Client) SendPacket(packet *protocol7.Packet) error {
 	// }
 
 	for _, callback := range client.Callbacks.PacketOut {
-		if callback(packet) == false {
+		if !callback(packet) {
 			return nil
 		}
 	}
 
 	client.LastSend = time.Now()
-	client.Conn.Write(packet.Pack(&client.Session))
+
+	data := packet.Pack(&client.Session)
+	n := 0
+	for len(data) > 0 {
+		n, err = client.Conn.Write(data[n:])
+		if err != nil {
+			return err
+		}
+		data = data[n:]
```

Comment:
what is happening here? Why is the data split up?

Reply:
it's just a safety measure, probably not necessary.
In case `Write` reports that it sent less data than you provided, I would like to continue Writing to the network connection until the whole data slice is consumed.

Reply:
wait a minute

Reply:
Not sure if I like that complexity without an actual use case. Do you know when that can happen? If that happens can it happen so many times that it becomes an infinite loop? Should we log something to the user when it happens? Isn't the go standard library or the os covering that case already?

What is your motivation for adding this?

Reply:
I think there are some limitations under the hood, (I might be wrong). Given enough data, this might actually return n < len(data).

Reply:
The teeworlds protocol also has a max payload size. That should be way below that limit.

Reply:
We should just check if data exceeds that maximum payload size and then throw an error instead of sending it.

Reply:
ok, I asked in the Go discord, seemingly that case will not happen, as Write is not allowed to partially write in case that there was no error returned. In case of an error, it might write partial data.

Reply:
fixed